### PR TITLE
{dev} [dummy] intel/2017.00 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**
Installer currently doesn't work (invalid components).

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 please mark this `(WIP)`

**Comment by bartoldeman** [comment]
The quotes needs to be removed in silent.cfg. I'll submit an easyblock PR.

**Comment by wpoely86** [comment]
@bartoldeman already on it. Intel support told us it's changed. But there is more, sanity check fails with missing symlink and some other minor details.

**Comment by bartoldeman** [comment]
Thanks. icc installation was ok here (with the quote removal in intelbase.py) but will see further.

**Comment by petruccigp** [comment]
Hi All,
Any updates on this issue? I'm facing exactly what you're mentioning. Any timeline where you guys believe intel 2017 will be fully working?
Thanks!

**Comment by bartoldeman** [comment]
Actually after the quote removal the whole thing appeared to have installed fine for me. I did have to change versions at a few places though:
imkl: 2017.0.098
impi: 2017.0.098
instead of 2017.0.98.

And the following small patches:
for impi:
```
@@ -1,7 +1,7 @@
 # This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
 
 name = 'impi'
-version = '2017.0.98'
+version = '2017.0.098'
 
 homepage = 'http://software.intel.com/en-us/intel-mpi-library/'
 description = """The Intel(R) MPI Library for Linux* OS is a multi-fabric message
@@ -10,7 +10,7 @@
 
 toolchain = {'name': 'iccifort', 'version': '2017.0.35-GCC-5.4.0-2.26'}
 
-sources = ['l_mpi_p_%(version)s.tgz']
+sources = ['l_mpi_%(version)s.tgz']
 
 checksums = ['fc123875773816b7084a91e419d54d20']
 
```
for imkl:
```
@@ -1,7 +1,7 @@
 # This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
 
 name = 'imkl'
-version = '2017.0.98'
+version = '2017.0.098'
 
 homepage = 'http://software.intel.com/en-us/intel-mkl/'
 description = """Intel Math Kernel Library is a library of highly optimized,
@@ -22,11 +22,13 @@
 
 postinstallcmds = [
     # extract the examples
-    'tar xvzf %(installdir)s/mkl/examples/examples_cluster.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_cluster_c.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_cluster_f.tgz -C %(installdir)s/mkl/examples/',
     'tar xvzf %(installdir)s/mkl/examples/examples_core_c.tgz -C %(installdir)s/mkl/examples/',
     'tar xvzf %(installdir)s/mkl/examples/examples_core_f.tgz -C %(installdir)s/mkl/examples/',
     'tar xvzf %(installdir)s/mkl/examples/examples_f95.tgz -C %(installdir)s/mkl/examples/',
-    'tar xvzf %(installdir)s/mkl/examples/examples_mic.tgz -C %(installdir)s/mkl/examples/'
+    'tar xvzf %(installdir)s/mkl/examples/examples_mic_c.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_mic_f.tgz -C %(installdir)s/mkl/examples/'
 ]
 
 modextravars = {
```

**Comment by akesandgren** [comment]
@wpoely86 Did you create an easyblock PR for this? I can't find any. If not i have the required patches including sanity check fix for mkl ready.

**Comment by boegel** [comment]
@damianam you mentioned running into some trouble when trying to install these, because of missing required changes to the easyblock?

@akesandgren I believe you mentioned you already succeeded in installing these, can you share what you had to tweak to make it work?

**Comment by akesandgren** [comment]
Yes i have easyblock changes available for 2017, i was just waiting for @wpoely86 to push his PR for it to see if they matched.

**Comment by boegel** [comment]
@akesandgren it may be quicker to open your PR, and ping @wpoely86 to review it ;)

**Comment by akesandgren** [comment]
Added easyblock PR 1012

File: easybuild/easyconfigs/i/impi/impi-2017.0.98-iccifort-2017.0.35-GCC-5.4.0-2.26.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This is an easyconfig file for EasyBuild, see http://hpcugent.github.io/easybuild
+
+name = 'impi'
+version = '2017.0.98'
```
Comment (boegel) [inline easybuild/easyconfigs/i/impi/impi-2017.0.98-iccifort-2017.0.35-GCC-5.4.0-2.26.eb]: @wpoely86 upstream version number for latest `impi` is `2017.0.098`?

cc @akesandgren

Comment (akesandgren) [inline easybuild/easyconfigs/i/impi/impi-2017.0.98-iccifort-2017.0.35-GCC-5.4.0-2.26.eb]: Yes intel changed all version numbers to match the compiler.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 0 out of 8 (8 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/5ff153d49e744b82593c585aec09b68d for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/bd3e50af4ce8b77b4a88892740e1249f for a full test report.

File: easybuild/easyconfigs/i/imkl/imkl-2017.0.098-iimpi-2017.00-GCC-5.4.0-2.26.eb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+postinstallcmds = [
+    # extract the examples
+    'tar xvzf %(installdir)s/mkl/examples/examples_cluster_c.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_cluster_f.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_core_c.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_core_f.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_f95.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_mic_f.tgz -C %(installdir)s/mkl/examples/',
+    'tar xvzf %(installdir)s/mkl/examples/examples_mic_c.tgz -C %(installdir)s/mkl/examples/',
```
Comment (boegel) [inline easybuild/easyconfigs/i/imkl/imkl-2017.0.098-iimpi-2017.00-GCC-5.4.0-2.26.eb]: @wpoely86 shouldn't we let the `imkl` easyblock deal with this instead?

**Comment by jhein32** [comment]
Is there any energy left regarding other components of the Intel cluster edition?  Things I am interested in are e.g. vtune, advisor inspector and also they make a bit of noise about their python (yet another python install).  

I am asking to avoid duplicated effort.

**Comment by bartoldeman** [comment]
I'll create a PR for VTune. I have the easyconfig sitting here.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2656.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7f84fcd9f68fc5d0e469234d7a286ad3 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
node2135.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/f7d24de5b50f5ded09af0bac92c67a55 for a full test report.

**Comment by boegel** [comment]
@wpoely86 please update PR title?

lgtm, we can handle the `imkl` examples in the easyblock later, if we care enough ;)

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**